config/config-api/src/main/java/com/thoughtworks/go/domain/packagerepository/PackageDefinition.java

```diff
@@ -326,4 +322,22 @@ public void ensureIdExists() {
             setId(UUID.randomUUID().toString());
         }
     }
+
+    @Override
+    public boolean hasSecretParams() {
+        return this.getRepository().hasSecretParams()
+                || this.getConfiguration()
+                .stream()
+                .anyMatch(ConfigurationProperty::hasSecretParams);
```

Comment:
The logic for `hasSecretParams` and `getSecretParams` is repeated across classes. Is it ok to move it to `Configuration` class?